sktime/datatypes/_check.py

```diff
@@ -237,18 +237,18 @@ def check_raise(obj, mtype: str, scitype: str = None, var_name: str = "input"):
     ValueError if mtype input argument is not of expected type
     """
     obj_long_name_for_avoiding_linter_clash = obj
-    valid, msg, _ = check_is_mtype(
+    valid = check_is_mtype(
         obj=obj_long_name_for_avoiding_linter_clash,
         mtype=mtype,
         scitype=scitype,
-        return_metadata=True,
+        return_metadata=False,
         var_name=var_name,
     )
 
     if valid:
         return True
     else:
-        raise TypeError(msg)
+        raise TypeError(f"Expected object mtype to be one of {list(mtype)}, was {type(obj)}")
```

Comment:
Copied from the previous discussion in https://github.com/sktime/sktime/pull/4140

> @[fkiraly](https://github.com/fkiraly) fkiraly [2 days ago](https://github.com/sktime/sktime/pull/4140/files#r1092270788)
> this message would be confusing, as some mtypes are called like class types, e.g., pd.DataFrame.